tools/libs/light/libxl_arm.c

```diff
@@ -949,19 +949,38 @@ static int make_vpci_node_common(libxl__gc *gc, void *fdt,
 }
 
 #define PCI_NUM_IRQ   4
-#define PCI_IRQ_MAP_STRIDE   8
+#define PCI_IRQ_MAP_MIN_STRIDE   8
 #define PCI_IOMMU_MAP_STRIDE 4
 
 static int create_vpci_irq_map(libxl__gc *gc, void *fdt)
 {
-    uint32_t full_irq_map[PCI_NUM_IRQ * PCI_NUM_IRQ * PCI_IRQ_MAP_STRIDE] = {0};
-    uint32_t *irq_map = full_irq_map;
+    uint32_t *full_irq_map, *irq_map;
+    size_t full_irq_map_sz;
     unsigned int slot, pin;
-    int res;
+    int res, cells;
 
     res = fdt_property_cell(fdt, "#interrupt-cells", 1);
     if (res) return res;
 
+    res = fdt_node_offset_by_phandle(fdt, GUEST_PHANDLE_GIC);
```

Comment:
So the code here now assumes that GIC node is already present. I would probably add a comment mentioning that.

Reply:
done